javax-money/src/main/java/com/fasterxml/jackson/datatype/money/MonetaryAmountDeserializer.java

```diff
@@ -0,0 +1,77 @@
+package com.fasterxml.jackson.datatype.money;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import com.fasterxml.jackson.databind.jsontype.TypeDeserializer;
+
+import javax.annotation.Nullable;
+import javax.money.CurrencyUnit;
+import javax.money.MonetaryAmount;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Arrays;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+import static java.lang.String.format;
+
+final class MonetaryAmountDeserializer<M extends MonetaryAmount> extends JsonDeserializer<M> {
+
+    private final MonetaryAmountFactory<M> factory;
+    private final FieldNames names;
+
+    MonetaryAmountDeserializer(final MonetaryAmountFactory<M> factory, final FieldNames names) {
+        this.factory = factory;
+        this.names = names;
+    }
+
+    @Override
+    public Object deserializeWithType(final JsonParser parser, final DeserializationContext context,
```

Comment:
If extending `StdDeserializer` (or, `StdScaralDeserializer`), wouldn't need to implement this method, I think.